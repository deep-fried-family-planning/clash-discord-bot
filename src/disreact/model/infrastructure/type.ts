
export type IsEqual<A, B> =
  A extends B
  ? true
  : false;

export type IsAny<A> =
  boolean extends (IsEqual<A, never>)
  ? true
  : false;

export type IfAny<A, B, C> =
  IsAny<A> extends true
  ? B
  : C;

// eslint-disable-next-line @typescript-eslint/no-unsafe-function-type
export type Fun = Function;

export type FnN<A extends unknown[], B> = (...p: A) => B;

type GetAsync<A> = Extract<A, Promise<any>>;

const __async = async () => {};

export const isAsync = <A extends unknown[], B>(u: FnN<A, B>): u is FnN<A, Extract<B, Promise<any>>> =>
  u.constructor === __async.constructor;
