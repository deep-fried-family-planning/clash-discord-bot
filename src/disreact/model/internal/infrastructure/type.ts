import type * as E from 'effect/Effect';

export type IsEqual<A, B> =
  A extends B
  ? true
  : false;

export type IsAny<A> =
  boolean extends (IsEqual<A, never>)
  ? true
  : false;

export type IfAny<A, B, C> =
  IsAny<A> extends true
  ? B
  : C;

// eslint-disable-next-line @typescript-eslint/no-unsafe-function-type
export type Fn = Function;

export type FnN<A extends any[], B> = (...p: A) => B;

type GetAsync<A> = Extract<A, Promise<any>>;

const __async = async () => {};

export const isAsync = <A extends any[], B>(u: FnN<A, B>): u is FnN<A, Extract<B, Promise<any>>> =>
  u.constructor === __async.constructor;

export type DataFirst<A, B extends any[], C> = (data: A, ...p: B) => C;

export type DataLast<A, B extends any[], C> = (...p: B) => (data: A) => C;

export type DataLastFromFirst<A> =
  A extends DataFirst<infer A, infer B, infer C>
  ? DataLast<A, B, C>
  : never;

export type AnyEffect = E.Effect<any, any, any>;

export type type = never;

export type UnifyM<A extends unknown[]> =
  A[number] extends AnyEffect
  ? E.Effect.AsEffect<A[number]>
  : A[number];
