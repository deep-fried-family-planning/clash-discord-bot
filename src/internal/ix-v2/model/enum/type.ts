import {Discord} from 'dfx';


export const ACTION_ROW = Discord.ComponentType.ACTION_ROW;
export const TEXT_INPUT = Discord.ComponentType.TEXT_INPUT;
export const USER_SELECT = Discord.ComponentType.USER_SELECT;
export const CHANNEL_SELECT = Discord.ComponentType.CHANNEL_SELECT;
export const ROLE_SELECT = Discord.ComponentType.ROLE_SELECT;
export const MENTIONABLE_SELECT = Discord.ComponentType.MENTIONABLE_SELECT;
export const STRING_SELECT = Discord.ComponentType.STRING_SELECT;


export const BUTTON = Discord.ComponentType.BUTTON;
export const BUTTON_TOGGLE = 15;
export const BUTTON_SUBMIT = 16;
export const BUTTON_DELETE = 17;
