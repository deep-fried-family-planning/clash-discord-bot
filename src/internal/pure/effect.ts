import {Order} from 'effect';


export {
    Effect as E,
    Console as CSL,
    Data as D,
    DateTime as DT,
    Redacted as RDT,
    Config as CFG,
    Context as CTX,
    Layer as L,
    Cache as C,
    Logger as Logger,
    LogLevel,
    LogSpan,
    Schema as S,
    Brand as B,
    RateLimiter as RL,
    Metric as Metric,
    Order as ORD,
    Option as O,
    Equal as EQ,

    Cron,

    pipe,
    pipe as p,
    flow,
    flow as f,

} from 'effect';


export {
    gen as g,

} from 'effect/Effect';


export const ORDS = Order.string;
export const ORDSR = Order.reverse(Order.string);
export const ORDN = Order.number;
export const ORDNR = Order.reverse(Order.number);


export * as Ar from 'effect/Array'; // collection array
export * as Kv from 'effect/Record'; // collection key-value
