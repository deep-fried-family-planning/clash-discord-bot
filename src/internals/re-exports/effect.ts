export {
    Effect as E,

    Config as Cfg,
    Redacted as Rdt,

    Console as CSL,

    Data as D,
    DateTime as DT,

    Redacted as RDT,
    Config as CFG,
    Context as CTX,
    Layer as L,

    Cache as C,

    Logger as Logger,
    LogLevel,
    LogSpan,

    Schema as S,

    Brand as B,

    RateLimiter as RL,

    Metric as Metric,

    flow,
    pipe,

} from 'effect';
