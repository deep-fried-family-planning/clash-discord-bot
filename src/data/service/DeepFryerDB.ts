import {DynamoEnv} from '#config/aws.ts';
import {DeepFryerDBCapacity} from '#src/data/service/DeepFryerDBCapacity.ts';
import type {DeleteCommandInput, GetCommandInput, PutCommandInput, QueryCommandInput, UpdateCommandInput} from '@aws-sdk/lib-dynamodb';
import {DynamoDBDocument} from '@effect-aws/lib-dynamodb';
import * as E from 'effect/Effect';
import * as Data from 'effect/Data';

export class DeepFryerDBError extends Data.TaggedError('DeepFryerDBError')<{
  cause: any;
}> {}

export class DeepFryerDB extends E.Service<DeepFryerDB>()('deepfryer/DB', {
  effect: E.gen(function* () {
    const env = yield* DynamoEnv;
    const document = yield* DynamoDBDocument;
    const capacity = yield* DeepFryerDBCapacity;

    return {
      put: (cmd: Partial<PutCommandInput>) =>
        document.put(
          {
            ...cmd,
            TableName: cmd.TableName ?? env.DFFP_DDB_TABLE,
          } as any,
        ).pipe(
          E.catchAll((cause) => new DeepFryerDBError({cause})),
          capacity.encodedWriteUnits(cmd.Item),
        ),

      get: (cmd: Partial<GetCommandInput>) =>
        document.get(
          {
            ...cmd,
            TableName: cmd.TableName ?? env.DFFP_DDB_TABLE,
          } as any,
        ).pipe(
          E.catchAll((cause) => new DeepFryerDBError({cause})),
          capacity.readLimiter,
        ),

      update: (cmd: Partial<UpdateCommandInput>) =>
        document.update(
          {
            ...cmd,
            TableName: cmd.TableName ?? env.DFFP_DDB_TABLE,
          } as any,
        ).pipe(
          E.catchAll((cause) => new DeepFryerDBError({cause})),
          capacity.writeLimiter,
        ),

      delete: (cmd: Partial<DeleteCommandInput>) =>
        document.delete(
          {
            ...cmd,
            TableName: cmd.TableName ?? env.DFFP_DDB_TABLE,
          } as any,
        ).pipe(
          E.catchAll((cause) => new DeepFryerDBError({cause})),
          capacity.writeLimiter,
        ),

      query: (cmd: Partial<QueryCommandInput>) =>
        document.query(
          {
            ...cmd,
            TableName: cmd.TableName ?? env.DFFP_DDB_TABLE,
          } as any,
        ).pipe(
          E.catchAll((cause) => new DeepFryerDBError({cause})),
          capacity.readLimiter,
        ),

      scan: (cmd: Partial<QueryCommandInput>) =>
        document.scan(
          {
            ...cmd,
            TableName: cmd.TableName ?? env.DFFP_DDB_TABLE,
          } as any,
        ).pipe(
          E.catchAll((cause) => new DeepFryerDBError({cause})),
          capacity.readLimiter,
        ),
    };
  }),
  dependencies: [
    DynamoDBDocument.defaultLayer,
    DeepFryerDBCapacity.Default,
  ],
  accessors: true,
}) {}
