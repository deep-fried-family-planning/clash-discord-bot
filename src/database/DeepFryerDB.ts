import {Codec, Server, ServerClan, UserPlayer} from '#src/database/arch/codec.ts';
import {GSI_ALL_CLANS, GSI_ALL_PLAYERS, GSI_ALL_SERVERS} from '#src/database/arch/const/gsi-tag.ts';
import {DeepFryerDocument} from '#src/database/service/DeepFryerDocument.ts';
import {DeepFryerPage} from '#src/database/service/DeepFryerPage.ts';
import {E, pipe} from '#src/internal/pure/effect.ts';
import * as Array from 'effect/Array';
import * as Chunk from 'effect/Chunk';
import * as Stream from 'effect/Stream';

export const decodeUpgradeItem = <A extends Codec>(s: A, item: Codec.Enc<A>) =>
  pipe(
    Codec.decodeItem(s, item),
    E.tap((dec) =>
      dec?.upgraded
        ? E.fork(saveItem(s, dec))
        : undefined,
    ),
  );

export const decodeUpgradeItems = <A extends Codec>(s: A, items: Chunk.Chunk<Codec.Enc<A>>) =>
  pipe(
    Chunk.toArray(items),
    Array.map((item) =>
      decodeUpgradeItem(s, item).pipe(E.catchTag('DataDecodeError', () => E.succeed(undefined))),
    ),
    E.allWith({concurrency: 'unbounded'}),
    E.map((items) => items.filter((item) => !!item)),
  );

export const saveItem = <A extends Codec>(s: A, item: A['_']) =>
  pipe(
    Codec.encodeItem(s, item),
    // E.tap((enc) => DeepFryerDocument.put(enc as any)),
    // E.tap((enc) => Console.log(enc)),
  );

export const readItem = <A extends Codec>(s: A, pk: string, sk: string) =>
  pipe(
    DeepFryerDocument.get({
      codec: s,
      Key  : s.encodeKey(pk, sk),
    }),
    E.flatMap((item) => E.fromNullable(item)),
    E.flatMap((item) => decodeUpgradeItem(s, item.Item as any)),
  );

export const readItem2 = <A extends Codec>(s: A, {pk, sk}: {pk: string; sk: string}) =>
  pipe(
    DeepFryerDocument.get({
      codec: s,
      Key  : s.encodeKey(pk, sk),
    }),
    E.flatMap((item) => E.fromNullable(item)),
    E.flatMap((item) => decodeUpgradeItem(s, item.Item as any)),
  );

export const readPartition = <A extends Codec>(s: A, pk: string) =>
  pipe(
    DeepFryerPage.pageQuery({
      KeyConditionExpression   : 'pk = :pk',
      ExpressionAttributeValues: {':pk': s.encodePk(pk)},
      ConsistentRead           : true,
      Limit                    : 25,
    }),
    E.flatMap((stream) => Stream.runCollect(stream)),
    E.flatMap((items) => decodeUpgradeItems(s, items as any)),
  );
export const readPartition2 = <A extends Codec>(s: A, {pk}: {pk: string}) =>
  pipe(
    DeepFryerPage.pageQuery({
      KeyConditionExpression   : 'pk = :pk',
      ExpressionAttributeValues: {':pk': s.encodePk(pk)},
      ConsistentRead           : true,
      Limit                    : 25,
    }),
    E.flatMap((stream) => Stream.runCollect(stream)),
    E.flatMap((items) => decodeUpgradeItems(s, items as any)),
  );

export const deleteItem = <A extends Codec>(s: A, pk: string, sk: string) =>
  DeepFryerDocument.delete({
    codec: s,
    Key  : s.encodeKey(pk, sk),
  });
export const deleteItem2 = <A extends Codec>(s: A, {pk, sk}: {pk: string; sk: string}) =>
  DeepFryerDocument.delete({
    codec: s,
    Key  : s.encodeKey(pk, sk),
  });

export const getClansForServer = (pk: string) =>
  readPartition(ServerClan, pk);

export const queryServerClans = (sk: string) =>
  pipe(
    DeepFryerPage.pageQuery({
      IndexName                : GSI_ALL_CLANS,
      KeyConditionExpression   : 'gsi_clan_tag = :gsi_clan_tag',
      ExpressionAttributeValues: {
        ':gsi_clan_tag': ServerClan.encodeSk(sk),
      },
    }),
    E.flatMap((stream) => Stream.runCollect(stream)),
    E.flatMap((items) => decodeUpgradeItems(ServerClan, items as any)),
  );

export const scanServerClans = () =>
  pipe(
    DeepFryerPage.pageScan({
      IndexName: GSI_ALL_CLANS,
    }),
    E.flatMap((stream) => Stream.runCollect(stream)),
    E.flatMap((items) => decodeUpgradeItems(ServerClan, items as any)),
  );

export const getPlayersForServer = (pk: string) =>
  readPartition(UserPlayer, pk);

export const queryUserPlayers = (sk: string) =>
  pipe(
    DeepFryerPage.pageQuery({
      IndexName                : GSI_ALL_PLAYERS,
      KeyConditionExpression   : 'gsi_player_tag = :gsi_player_tag',
      ExpressionAttributeValues: {
        ':gsi_player_tag': UserPlayer.encodeSk(sk),
      },
    }),
    E.flatMap((stream) => Stream.runCollect(stream)),
    E.flatMap((items) => decodeUpgradeItems(UserPlayer, items as any)),
  );

export const scanUserPlayers = () =>
  pipe(
    DeepFryerPage.pageScan({
      IndexName: GSI_ALL_PLAYERS,
    }),
    E.flatMap((stream) => Stream.runCollect(stream)),
    E.flatMap((items) => decodeUpgradeItems(UserPlayer, items as any)),
  );

export const scanServers = () =>
  pipe(
    DeepFryerPage.pageScan({
      IndexName: GSI_ALL_SERVERS,
    }),
    E.flatMap((stream) => Stream.runCollect(stream)),
    E.flatMap((items) => decodeUpgradeItems(Server, items as any)),
  );
