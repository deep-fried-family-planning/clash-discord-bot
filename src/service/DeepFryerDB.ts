import {DynamoEnv} from '#config/aws.ts';
import {DeepFryerDBCapacity} from '#src/service/DeepFryerDBCapacity.ts';
import type {DeleteCommandInput, GetCommandInput, PutCommandInput, QueryCommandInput, UpdateCommandInput} from '@aws-sdk/lib-dynamodb';
import {DynamoDBDocument} from '@effect-aws/lib-dynamodb';
import * as E from 'effect/Effect';

export class DeepFryerDB extends E.Service<DeepFryerDB>()('deepfryer/DB', {
  effect: E.gen(function* () {
    const env = yield* DynamoEnv;
    const document = yield* DynamoDBDocument;
    const capacity = yield* DeepFryerDBCapacity;

    return {
      put: (cmd: Partial<PutCommandInput>) =>
        document.put(
          {
            ...cmd,
            TableName: cmd.TableName ?? env.DFFP_DDB_TABLE,
          } as any,
        ).pipe(
          capacity.encodedWriteUnits(cmd.Item),
        ),

      get: (cmd: Partial<GetCommandInput>) =>
        document.get(
          {
            ...cmd,
            TableName: cmd.TableName ?? env.DFFP_DDB_TABLE,
          } as any,
        ).pipe(
          capacity.readLimiter,
        ),

      update: (cmd: Partial<UpdateCommandInput>) =>
        document.update(
          {
            ...cmd,
            TableName: cmd.TableName ?? env.DFFP_DDB_TABLE,
          } as any,
        ).pipe(
          capacity.writeLimiter,
        ),

      delete: (cmd: Partial<DeleteCommandInput>) =>
        document.delete(
          {
            ...cmd,
            TableName: cmd.TableName ?? env.DFFP_DDB_TABLE,
          } as any,
        ).pipe(
          capacity.writeLimiter,
        ),

      query: (cmd: Partial<QueryCommandInput>) =>
        document.query(
          {
            ...cmd,
            TableName: cmd.TableName ?? env.DFFP_DDB_TABLE,
          } as any,
        ).pipe(
          capacity.readLimiter,
        ),

      scan: (cmd: Partial<QueryCommandInput>) =>
        document.scan(
          {
            ...cmd,
            TableName: cmd.TableName ?? env.DFFP_DDB_TABLE,
          } as any,
        ).pipe(
          capacity.readLimiter,
        ),
    };
  }),
  dependencies: [
    DynamoDBDocument.defaultLayer,
    DeepFryerDBCapacity.Default,
  ],
  accessors: true,
}) {}
