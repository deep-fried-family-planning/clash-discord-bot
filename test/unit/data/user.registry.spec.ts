import {DataTag} from '#src/data/constants/index.ts';
import {type User, UserRegistry} from '#src/data/index.ts';
import {DeepFryerDB} from '#src/service/DeepFryerDB.ts';
import type {DeleteCommandInput, GetCommandInput, PutCommandInput, QueryCommandInput, ScanCommandInput} from '@aws-sdk/lib-dynamodb';
import {it, vi} from '@effect/vitest';
import * as DateTime from 'effect/DateTime';
import * as E from 'effect/Effect';
import {pipe} from 'effect/Function';
import * as L from 'effect/Layer';

const mock = {
  get   : vi.fn((cmd: GetCommandInput) => E.succeed({Item: {} as User.Encoded | undefined})),
  put   : vi.fn((cmd: PutCommandInput) => E.void),
  delete: vi.fn((cmd: DeleteCommandInput) => E.void),
  query : vi.fn((cmd: QueryCommandInput) => E.void),
  scan  : vi.fn((cmd: ScanCommandInput) => E.void),
};

const mockDb = L.succeed(DeepFryerDB, DeepFryerDB.make(mock as any));

it.effect('when registering a new user', E.fn(function* () {
  mock.get.mockReturnValueOnce(E.succeed({Item: undefined}));

  const actual = yield* pipe(
    UserRegistry.register({
      caller_id: 'user',
      payload  : {
        timezone: DateTime.zoneUnsafeMakeNamed('America/Chicago'),
      },
    }),
    E.provide(mockDb),
  );

  expect(actual).toMatchSnapshot();
  expect(mock.get.mock.calls[0][0].Key).toMatchSnapshot();
  expect(mock.put.mock.calls[0][0].Item).toMatchSnapshot();
}));

it.effect('when re-registering a user', E.fn(function* () {
  mock.get.mockReturnValueOnce(E.succeed({
    Item: {
      _tag           : DataTag.USER,
      pk             : 'u-user',
      sk             : 'now',
      version        : 0,
      created        : DateTime.unsafeMake(0).pipe(DateTime.format()),
      updated        : DateTime.unsafeMake(0).pipe(DateTime.format()),
      gsi_all_user_id: 'u-user',
      timezone       : 'America/New_York',
    },
  }));

  const actual = yield* pipe(
    UserRegistry.register({
      caller_id: 'user',
      payload  : {
        timezone: DateTime.zoneUnsafeMakeNamed('America/Chicago'),
      },
    }),
    E.provide(mockDb),
  );

  expect(actual).toMatchSnapshot();
  expect(mock.get.mock.calls[0][0].Key).toMatchSnapshot();
  expect(mock.put.mock.calls[0][0].Item).toMatchSnapshot();
}));

it.effect('when admin registering a new user', E.fn(function* () {
  mock.get
    .mockReturnValueOnce(E.succeed({
      Item: {
        _tag           : DataTag.USER,
        pk             : 'u-user1',
        sk             : 'now',
        version        : 0,
        created        : DateTime.unsafeMake(0).pipe(DateTime.format()),
        updated        : DateTime.unsafeMake(0).pipe(DateTime.format()),
        gsi_all_user_id: 'u-user1',
        timezone       : 'America/New_York',
      },
    }))
    .mockReturnValueOnce(E.succeed({Item: undefined}));

  const actual = yield* pipe(
    UserRegistry.register({
      caller_id: 'user1',
      target_id: 'user2',
      payload  : {
        timezone: DateTime.zoneUnsafeMakeNamed('America/Chicago'),
      },
    }),
    E.provide(mockDb),
  );

  expect(actual).toMatchSnapshot();
  expect(mock.get.mock.calls[0][0].Key).toMatchSnapshot();
  expect(mock.put.mock.calls[0][0].Item).toMatchSnapshot();
}));

it.effect('when admin registering a current user', E.fn(function* () {
  mock.get
    .mockReturnValueOnce(E.succeed({
      Item: {
        _tag           : DataTag.USER,
        pk             : 'u-user1',
        sk             : 'now',
        version        : 0,
        created        : DateTime.unsafeMake(0).pipe(DateTime.format()),
        updated        : DateTime.unsafeMake(0).pipe(DateTime.format()),
        gsi_all_user_id: 'u-user1',
        timezone       : 'America/New_York',
      },
    }))
    .mockReturnValueOnce(E.succeed({
      Item: {
        _tag           : DataTag.USER,
        pk             : 'u-user2',
        sk             : 'now',
        version        : 0,
        created        : DateTime.unsafeMake(0).pipe(DateTime.format()),
        updated        : DateTime.unsafeMake(0).pipe(DateTime.format()),
        gsi_all_user_id: 'u-user2',
        timezone       : 'America/New_York',
      },
    }));

  const actual = yield* pipe(
    UserRegistry.register({
      caller_id: 'user1',
      target_id: 'user2',
      payload  : {
        timezone: DateTime.zoneUnsafeMakeNamed('America/Chicago'),
      },
    }),
    E.provide(mockDb),
    E.catchAll((cause) => E.succeed(cause)),
  );

  expect(actual).toMatchSnapshot();
  expect(mock.get.mock.calls[0][0].Key).toMatchSnapshot();
  expect(mock.put).toBeCalledTimes(0);
}));
