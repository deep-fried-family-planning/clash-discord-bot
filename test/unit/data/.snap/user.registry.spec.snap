// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`when admin registering a current user 1`] = `[RegistryAdminError: User already registered.]`;

exports[`when admin registering a current user 2`] = `
{
  "pk": "u-user1",
  "sk": "now",
}
`;

exports[`when admin registering a new user 1`] = `
{
  "description": "Success",
}
`;

exports[`when admin registering a new user 2`] = `
{
  "pk": "u-user1",
  "sk": "now",
}
`;

exports[`when admin registering a new user 3`] = `
{
  "_tag": "User",
  "created": "1970-01-01T00:00:00.000Z",
  "gsi_all_user_id": "u-user2",
  "pk": "u-user2",
  "sk": "now",
  "timezone": "America/Chicago",
  "updated": "1970-01-01T00:00:00.000Z",
  "version": 0,
}
`;

exports[`when re-registering a user 1`] = `
{
  "description": "Success",
}
`;

exports[`when re-registering a user 2`] = `
{
  "pk": "u-user",
  "sk": "now",
}
`;

exports[`when re-registering a user 3`] = `
{
  "_tag": "User",
  "created": "1970-01-01T00:00:00.000Z",
  "gsi_all_user_id": "u-user",
  "pk": "u-user",
  "sk": "now",
  "timezone": "America/Chicago",
  "updated": "1970-01-01T00:00:00.000Z",
  "version": 0,
}
`;

exports[`when registering a new user 1`] = `
{
  "description": "Success",
}
`;

exports[`when registering a new user 2`] = `
{
  "pk": "u-user",
  "sk": "now",
}
`;

exports[`when registering a new user 3`] = `
{
  "_tag": "User",
  "created": "1970-01-01T00:00:00.000Z",
  "gsi_all_user_id": "u-user",
  "pk": "u-user",
  "sk": "now",
  "timezone": "America/Chicago",
  "updated": "1970-01-01T00:00:00.000Z",
  "version": 0,
}
`;
